### PR TITLE
 chore: unify common db ops under one trait 

### PR DESCRIPTION
**Pull request by dpc**
Looking at #2926 . Some initial cleanup.

### CODE REVIEW

File: fedimint-core/src/db/mod.rs [hunk 1]
```diff
@@ -339,6 +339,40 @@ impl Database {
     }
 }
 
+#[apply(async_trait_maybe_send!)]
+pub trait IDatabaseTransactionOps<'a>: 'a + MaybeSend {
```
Comment (dpc) [inline fedimint-core/src/db/mod.rs:343]: Basically I extracted this, so this methods can't diverge between multiple traits.

File: fedimint-rocksdb/src/lib.rs [hunk 2]
```diff
@@ -133,6 +135,26 @@ impl<'a> IDatabaseTransaction<'a> for RocksDbTransaction<'a> {
         }))
     }
 
+    async fn raw_remove_by_prefix(&mut self, key_prefix: &[u8]) -> anyhow::Result<()> {
+        fedimint_core::task::block_in_place(|| {
```
Comment (dpc) [inline fedimint-rocksdb/src/lib.rs:139]: @m1sterc001guy Notably, I was forced to implement it.

**Review by elsirion — DISMISSED** [review]
Great PR! I remember being annoyed about the duplication and not having the time to fix it.

**Comment by elsirion** [comment]
Looking at CI there seems to be a bug though … :(

**Comment by dpc** [comment]
@elsirion was missing range ending conditions in my implementation of delete_by_prefix for rocksdb.

File: fedimint-rocksdb/src/lib.rs (hunk 2 above)
Comment (m1sterc001guy) [reply fedimint-rocksdb/src/lib.rs:137]: We were relying on the default implementation before right?

Comment (m1sterc001guy) [reply fedimint-rocksdb/src/lib.rs:137]: This looks good to me, as long as `verify_remove_by_prefix` is passing I think we're good

**Review by m1sterc001guy — APPROVED** [review]
LGTM

File: fedimint-rocksdb/src/lib.rs (hunk 2 above)
Comment (dpc) [reply fedimint-rocksdb/src/lib.rs:137]: Yes, but that default impl was meh (loading everything into memory at once), and now it doesn't work due to object-safety anyway.

**Pull request merged**